test/framework/src/main/java/org/elasticsearch/cluster/node/TestDiscoveryNode.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.node;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Collections;
+
+public class TestDiscoveryNode {
```

Comment:
A class like `TestSomeClass` in our repo is often an implementation of `SomeClass`, so that's what I expected here. I'd think a class like this might be called `DiscoveryNodeTestUtils` or something, but that is more verbose, so I'm okay with the side of the trade-off you've picked here. We might want to rename the class later if more gets added to it than these static factory methods.